### PR TITLE
[RFC] Codemod invariant -> throw new Error

### PR DESCRIPTION
**Pull request by acdlite**
## Based on #22428 and #22429

*[Submitting for discussion]*

A few years ago, I updated our build to compile `invariant` calls to throw expressions: #15071. We still use `invariant` in the source, though, because our error minification script depended on it.

Now that we can minify throw expressions, too, (#22428 and #22429), `invariant` is no longer needed.

This PR converts all uses of `invariant` to the equivalent `throw` call expression.

The commits are designed to be reviewed in sequence.

I wrote an automated codemod and used jscodeshift to perform the conversion. I checked the script in temporarily for reference, then reverted it in a later commit, since there's no reason to keep it around in the repo.

All the automated changes — the codemod itself, ESLint autofix to remove the `invariant` imports, and Prettier — are contained in a single commit. The few manual changes I made after that are contained in a subsequent commit.

One of the things the codemod does is simplify condition passed to invariant:

- `invariant(false, msg)` is converted to an unguarded throw statement instead of `if (!false) {...}`. 
- `invariant(a === b)` is converted to `if (a !== b) {...}`
- ...and so on

Comparisons involving `>`, `<`, etc cannot be safely inverted in all cases. For example, `!(NaN > foo)` is not the same as `!NaN <= foo`. In those cases, the codemod would output a FIXME comment to prompt a manual review. You can see those in the commit that contains the manual changes.

### CODE REVIEW

**Comment by acdlite** [comment]
The reason the Node build of react-fetch increased in size is because it contains a single error that previously wasn't being minified. But now it is: https://github.com/facebook/react/pull/22428#issuecomment-927188935

And because it's the only minified error in that package, pulling in `formatProdErrorMessage` increases the overall size because there aren't other errors to amortize the cost.

**Review by rickhanlonii — COMMENTED** [review]
I'm +1 but as mentioned on the other PR, it's seems hard to disambiguate between errors that are minified and one's that are ignored. What about either an opt-in, like `throw MinifiedError()` or opt-out with `throw UnminifiedError()`.

**Comment by acdlite** [comment]
Replied here: https://github.com/facebook/react/pull/22429#issuecomment-931466429

> [...] we're not super consistent about which errors we minify and which ones we don't, because there's no existing lint rule that enforces it either way.
>
> The first step should be to decide which packages should have minified error messages. I think probably everything that runs in the browser, but not stuff that runs elsewhere. For RN we don't minify currently, which I suppose makes sense since React itself is never downloaded OTA (I think?).
>
> If there are exceptions after that we can consider adding an escape hatch, though I would still prefer an inline comment directive (like `// no-minify-error-message` or something) since it doesn't have runtime implications and we can adapt it to work with any Error subclass, like TypeError for example.

File: scripts/rollup/bundles.js [hunk 1]
```diff
@@ -216,15 +228,18 @@ const bundles = [
     bundleTypes: [FB_WWW_DEV, NODE_DEV, NODE_PROD, UMD_DEV, UMD_PROD],
     entry: 'react-dom/test-utils',
     global: 'ReactTestUtils',
+    minifyWithProdErrorCodes: false,
     externals: ['react', 'react-dom'],
   },
 
+  // TODO: We don't use this bundle anymore
```
Comment (bvaughn) [inline scripts/rollup/bundles.js:235]: @acdlite What do you mean? The Test Selector API kind of stalled but to my knowledge, we still have plans to make use of it. (I'm thinking about potentially using it for DevTools e2e tests at the moment via #22760)

**Pull request merged**